crates/fluvio/src/producer/config.rs

```diff
@@ -118,6 +120,9 @@ pub struct TopicProducerConfig {
 
     #[builder(default)]
     pub(crate) smartmodules: Vec<SmartModuleInvocation>,
+
+    #[builder(setter(into, strip_option), default)]
+    pub(crate) callback: Option<SharedProducerCallback<ProduceCompletionEvent>>,
```

Comment:
why make generic? don't think there is need to make generic

Reply:
Good catch!
It was to make it easy to handle lifetimes in my first implementation. But it's not needed for actual implementation.

Reply:
fixed